File: Day-20/BinaryToDecimal.js
function BinaryToDecimal(n) {
  let answer = 0;
  for (let i = 0; n != 0; i++) {
    let digit = n % 10;
    if (digit === 1) {
      answer = answer + Math.pow(2, i);
    }
    n = Math.floor(n / 10);
  }
  console.log(answer);
}

BinaryToDecimal(101);

File: Day-20/BinaryToDecimal.py
import math
n = int(input())

def BinToDec(n):
    answer = 0
    i = 0

    while(n != 0):
        digit = n % 10
        if(digit == 1):
            answer = answer + math.pow(2, i)
        n = math.floor(n/10)
        i += 1
        
    print(math.floor(answer))

BinToDec(n)